### PR TITLE
Srtest

### PR DESCRIPTION
**Pull request by steveri**
Added gold-standard comparisons for intermediate files, and it seems to be running successfully see e.g. https://travis-ci.org/StanfordAHA/CGRAFlow/builds/275589583.

This new flow includes comparisons for three intermediate files, e.g. for file 'pointwise" it compares
  - Halide DAG: 'pointwise_design_top.json'
  - CoreIR DAG: 'pointwise_mapped.json'
  - PNR    DAG: 'pointwise_annotated'

Note the Halide output DAG is sufficiently deterministic that (for now at least) I can do a straight text-compare (diff) to check for correctness.

The other two comparisons are a kind of half-assed topological compare of my own invention, necessitated by the fact that, given the same input, the mapper and PNR tools will create nondeterministically  different (but correct) outputs.  [Update: Ross says mapper output is now deterministic; so I may change this to an exact-compre in a later revision]  The topo compare is not perfect, but good enough to have caught or most recent failed build, which was the goal...see me if you want details...

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -82,3 +82,27 @@ Halide_CoreIR/test/scripts/install_travis.sh
...
+# 
+# # export PKG_CONFIG_PATH=/home/PATH/TO/VIRT/lib/pkgconfig
+# export PKG_CONFIG_PATH=$VIRT/lib/pkgconfig
+# 
+# # ./configure --prefix=/home/PATH/TO/VIRT/
+# ./configure --prefix=$VIRT/
+# 
+# make 
+# make install
+#
```
Comment (leonardt) [inline install.sh]: Do we need to keep this commented code around? Would be okay if you're planning to integrate it later, but it seems like this is just an alternative way to build/install pytgtk (installed using apt above).

File: makefile [hunk 2]
```diff
@@ -98,9 +116,17 @@ build/%_design_top.json: %_input_image Halide_CoreIR/apps/coreir_examples/%
 	od -t u1 build/$*_input.raw      | head
 	od -t u1 build/$*_halide_out.raw | head
 
-	ls -la build
+        # pwd
+        # ls -la build
```
Comment (leonardt) [inline makefile]: Needed?

File: makefile [hunk 3]
```diff
@@ -98,9 +116,17 @@ build/%_design_top.json: %_input_image Halide_CoreIR/apps/coreir_examples/%
...
 
 	cat build/$*_design_top.json $(OUTPUT)
+
+	echo "GOLD-COMPARE --------------------------------------------------" \
+	  | tee -a test/compare_summary.txt
+	test/compare.csh $@ diff 2>&1 | tee -a test/compare_summary.txt
+
+#	make test/$@.compare
```
Comment (leonardt) [inline makefile]: `# make test/$@.compare`
Needed?

File: makefile [hunk 4]
```diff
@@ -114,9 +140,22 @@ build/%_mapped.json: build/%_design_top.json
 	@echo; echo Making $@ because of $?
 	echo "MAPPER"
 	./CGRAMapper/bin/map build/$*_design_top.json build/$*_mapped.json $(OUTPUT)
-	ls -la build
+        # ls -la build
```
Comment (leonardt) [inline makefile]: Needed?

File: tmp.force-rebuild [hunk 5]
```diff
@@ -0,0 +1 @@
+Fri Sep  8 12:14:41 PDT 2017
```
Comment (leonardt) [inline tmp.force-rebuild]: What is this file for?

**Comment by steveri** [comment]
Oof.  Lenny, thanks so much for your thoughtful review and reply.  Clearly I forgot to do a cleanup step before filing the git pull, something I usually do but somehow missed this time...I will delete a bunch of unnecessary crap before the final merge...

**Comment by steveri** [comment]
Okay, finished my code cleanup.  It's all comments and whitespace removal ish, so should not have affected functionality at all, we shall see after travis script final checks have completed.

**Comment by steveri** [comment]
PNR is having intermittent problems, is why the latest check failed (https://travis-ci.org/StanfordAHA/CGRAFlow/builds/277409056).  Nothing to do with my changes I think, but I am rerunning just in case.

**Comment by steveri** [comment]
It's been almost a week with no new complaints, I'm gonna go ahead and merge...

**Pull request merged**